### PR TITLE
Fix logical flow between the relying party components and other misc fixes

### LINKED ISSUE #105: The logical connection between components shown in relying party is incorrect in the architecture diagram
The two confidential-containers v1 diagrams in architecture [doc](https://github.com/confidential-containers/documentation/blob/main/architecture.md) incorrectly depict the `Attestation Service` as the remote attestation entry point.
Thanks to @dcmiddle for highlighting the same in the community slack channel.

### PR DESCRIPTION
**Pull request by bpradipt**
Fixes: #105

### CODE REVIEW

**Comment by mythi** [comment]
I started updating related docs in this repo and here's what I got:

```
diff --git a/architecture.md b/architecture.md
index bff2622..e6d17fa 100644
--- a/architecture.md
+++ b/architecture.md
@@ -196,8 +196,6 @@ Containers using VM-based TEEs and the Kata Containers runtime:
 ![COCO_ccv1_TEE](https://github.com/confidential-containers/documentation/blob/main/images/COCO_ccv1_TEE.jpg)
 
 The following diagram shows the upcoming v1 architecture to run Confidential
-Containers using the Intel SGX process-based TEE. It relies on the
-[Inclavare Containers](https://github.com/inclavare-containers/inclavare-containers)
-runtime:
+Containers using the Intel SGX process-based TEE:
```

Since you're updating the arch doc, maybe drop Inclavare from the diagrams too?

We don't rely on `rune`. We have a custom shim called `shim-rune` but it uses `runc`.

**Review by dcmiddle — APPROVED** [review]
Thanks for the fix!

File: images/COCO_ccv1_TEE.png
Comment (fitzthum) [inline images/COCO_ccv1_TEE.png:1]: I think both arrows to/from the Attestation Agent should go to the KBS. The Attestation Service never talks to the AA directly.

Comment (dcmiddle) [reply images/COCO_ccv1_TEE.png:1]: Oh true. I should have caught that.

Comment (bpradipt) [reply images/COCO_ccv1_TEE.png:1]: Right.. I'll fix it.

**Comment by bpradipt** [comment]
I updated the doc and all the figures, including the process-based TEE diagram and references. Also added a diagram for peer-pods approach.

@dcmiddle @sameo @mythi @fitzthum ptal

**Review by mythi — APPROVED** [review]
enclave-cc LGTM, thanks!

**Review by dcmiddle — APPROVED** [review]

**Review by fitzthum — APPROVED** [review]

**Pull request merged**